proposals/memory-specializations/utilities.tex

```diff
@@ -0,0 +1,636 @@
+  requires DefaultConstructible<value_type_t<I>>() &&
+           __ReferenceTo<I, value_type_t<I>>()
+    I uninitialized_value_construct(I first, S last);
+template <ForwardRange Rng>
+  requires DefaultConstructible<value_type_t<iterator_t<Rng>>>() &&
+           __ReferenceTo<iterator_t<Rng>, value_type_t<iterator_t<Rng>>>()
+    safe_iterator_t<Rng>
+    uninitialized_value_construct(Rng&& rng);
+template <ForwardIterator I>
+  requires DefaultConstructible<value_type_t<I>>() &&
+           __ReferenceTo<I, value_type_t<I>>()
+    I uninitialized_value_construct_n(I first, difference_type_t<I> n);
+template <InputIterator I, Sentinel<I> S, ForwardIterator O>
+  requires Constructible<value_type_t<O>, reference_t<I>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(I), tag::out(O)>
+    uninitialized_copy(I first, S last, O result);
+template <InputRange Rng, ForwardIterator O>
+  requires Constructible<value_type_t<O>, reference_t<iterator_t<Rng>>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(safe_iterator_t<Rng>), tag::out(O)>
+    uninitialized_copy(Rng&& rng, O result);
+template <InputIterator I, ForwardIterator O>
+  requires Constructible<value_type_t<O>, reference_t<I>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(I), tag::out(O)>
+    uninitialized_copy_n(I first, difference_type_t<I> n, O result);
+template <InputIterator I, Sentinel<I> S, ForwardIterator O>
+  requires Constructible<value_type_t<O>, rvalue_reference_t<I>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(I), tag::out(O)> uninitialized_move(I first, S last, O result);
+template <InputRange Rng, ForwardIterator O>
+  requires Constructible<value_type_t<O>, rvalue_reference_t<iterator_t<Rng>>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(safe_iterator_t<Rng>), tag::out(O)>
+    uninitialized_move(Rng&& rng, O result);
+template <InputIterator I, ForwardIterator O>
+  requires Constructible<value_type_t<O>, rvalue_reference_t<I>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(I), tag::out(O)>
+    uninitialized_move_n(I first, difference_type_t<I> n, O result);
+template <ForwardIterator I, Sentinel<I> S, class T>
+  requires Constructible<value_type_t<I>, const T&>() &&
+           __ReferenceTo<I, T>()
+  I uninitialized_fill(I first, S last, const T& x);
+template <ForwardRange Rng, class T>
+  requrires Constructible<value_type_t<iterator_t<Rng>>, const T&>() &&
+            __ReferenceTo<iterator_t<Rng>, T>()
+  safe_iterator_t<I>
+  uninitialized_fill(Rng&& rng, const T& x);
+template <ForwardIterator I, class T>
+  requires Constructible<value_type_t<I>, const T&>() &&
+           __ReferenceTo<I, T>()
+  I uninitialized_fill_n(I first, difference_type_t<I> n, const T& x);
+template <Destructible T>
+  void destroy_at(T* location) noexcept;
```

Comment:
I've added an overload (I don't know if removing the pointer version will upset too many people, and I can't think of any problems by having both...)